src/Store.js

```diff
@@ -580,31 +651,48 @@ class Store {
   /**
    * Gets records for type of collection from observable
    *
+   * NOTE: We only return records by unique id, this handles a scenario
+   * where the store keeps around a reference to a newly persisted record by its temp uuid.
+   * We can't simply remove the temp uuid reference because other
+   * related models may be still using the temp uuid in their relationships
+   * data object. However, when we are listing out records we want them
+   * to be unique by the persisted id (which is updated after a Model.save)
+   *
    * @method getRecords
    * @param {String} type
    * @return {Array} array of objects
    */
   getRecords (type) {
```

Comment:
I'm a little confused about what the difference is between this and `getAll`, other than `getAll` uses `getRecords`. Do we use `getRecords` a lot? Would it make sense to consolidate this with `getAll` since it's not a very long method? Or maybe just make it more explicitly a private method if it's only used in this file?

Reply:
getRecords is a small piece of code that just gets records with no interest in looking at the cache. We use it in portal and possibly mobile but I will be updating those to use `getAll`. We definitely need to simplify our getRecord/getCachedRecord/getRecords loopiness but I didn't address that with these changes